public/scripts/userpage.js

```diff
@@ -59,9 +69,83 @@ friendlyPix.UserPage = class {
         this.followingContainer.hide();
         this.nbFollowingContainer.removeClass('is-active');
       });
+
+      // Event bindings for Privacy Consent Dialog
+      this.privacyDialogButton.click(() => this.showPrivacyDialog());
+      this.privacyDialogSave.click(() => this.savePrivacySettings());
+      this.allowDataProcessing.change(() => {
+        // The submit button for the privacy settings is disabled until
+        // the data privacy policy is agreed to
+        if (this.allowDataProcessing.is(':checked')) {
+            this.privacyDialogSave.removeAttr("disabled");
+        }
+      });
     });
   }
 
+  /**
+   * Sets initial state of Privacy Dialog.
+   */
+  showPrivacyDialog() {
+    const userPromise = new Promise(function(resolve, reject) {
+      firebase.auth().onAuthStateChanged(function(user) {
+        if (user) {
+          resolve(user.uid);
+        } else {
+          reject();
+        }
+      });
+    }).then((uid) => {
+      this.userId = uid;
+      this.initializePrivacySettings(uid);
```

Comment:
Most of this method is a hack to set `this.userId` because I thought calling `loadUser` on `onAuthStateChanged` could have unintended side effects. Is there a better way to grab a `uid`? I just want to fetch the saved settings.